src/errors.rs

```diff
@@ -0,0 +1,31 @@
+use core::fmt;
+#[cfg(feature = "std")]
+use std::error::Error;
+
+const CAPERROR: &'static str = "insufficient capacity";
+
+/// Capacity error returned when container's capacity is not enough to complete the operation
+#[derive(Debug, PartialEq)]
+pub struct CapacityError {
+    /// Maximum characters allowed
+    pub maximum: usize,
+    /// Encountered number of characters
+    pub encountered: usize,
```

Comment:
It would be great if there was a field to return the data that was left over in the operation that overflowed. Like so:
```
struct CapacityError<T = ()> {
    pub maximum: usize,
    pub encountered: usize,
    pub rest: T 
}
```

What are your opinions on that one?